### PR TITLE
bugfix/acting-user-logic

### PR DESCRIPTION
**Pull request by SimRoberts9**
Added logic to check if the user is not null, then set 'actingAs' that user on the setActingUser function

### CODE REVIEW

File: src/app/Providers/LdapEmulatorServiceProvider.php [hunk 1]
```diff
@@ -37,9 +37,10 @@ public static function setActingUser(string $username)
         $ldapUsernameKey = config('ldap-emulator.ldap-username-key');
         $model = config('ldap-emulator.ldap-user-model');
 
-        Container::getDefaultConnection()->actingAs(
-            $model::findBy($ldapUsernameKey, strtolower($username)),
-        );
+        $user =  $model::findBy($ldapUsernameKey, strtolower($username));
```
Comment (AnthonyEdmonds) [inline src/app/Providers/LdapEmulatorServiceProvider.php]: Double space.

File: src/app/Providers/LdapEmulatorServiceProvider.php [hunk 2]
```diff
@@ -37,9 +37,10 @@ public static function setActingUser(string $username)
...
+        if ($user !== null) {
```
Comment (AnthonyEdmonds) [inline src/app/Providers/LdapEmulatorServiceProvider.php:41]: What happens if you throw an authentication exception at this point?

**Review by AnthonyEdmonds — CHANGES_REQUESTED** [review]

File: src/app/Providers/LdapEmulatorServiceProvider.php (hunk 2 above)
Comment (SimRoberts9) [reply src/app/Providers/LdapEmulatorServiceProvider.php:41]: I will give this a go on the Home laravel 11 branch

Comment (SimRoberts9) [reply src/app/Providers/LdapEmulatorServiceProvider.php:41]: I have put and AuthenticationException in the body of the if statement and the testPutsEmailInSessionOnSuccess and testRedirectsOnSuccess fails. I tried a few variations, but no joy

**Review by AnthonyEdmonds — APPROVED** [review]

**Pull request merged**